File: docs/JAMIA/comparisons/RPython/TM/25416955_string_dates_plot.py
# ID: SO#25416955
# Language: Python
# Bug class: Value type
# Runtime consequence: Crash
# In study: Yes
# Inclusion rationale: Matplotlib date axis from string column not parsed. String where date expected.
# Reproduction status: Live
# Type system catch: `Temporal` rejects `string`

import pandas as pd
import numpy as np

# Simulate data read from file with string dates
df = pd.DataFrame({
    'time': ['2014-07-10 11:49:14', '2014-07-10 11:50:14', '2014-07-10 11:51:14'],
    'amount': [45, 45, 21],
})

print(f"time dtype: {df['time'].dtype}")

# BUG: trying to do datetime arithmetic on strings
try:
    diff = df['time'].iloc[1] - df['time'].iloc[0]
    print(f"diff: {diff}")
except TypeError as e:
    print(f"Crash: {e}")

File: docs/JAMIA/comparisons/RPython/TM/25416955_string_dates_plot.ts
/**
 * RPython SO#25416955 — Plot pandas dates in matplotlib
 * Effect: Crash
 * Bug class: Type coercion
 *
 * In pandas, datetime column read from CSV as strings can't be used for
 * date arithmetic or time-axis plotting without explicit conversion.
 *
 * In tidy-ts, the column is typed as string. Attempting numeric operations
 * (which would be needed for date arithmetic) is caught at compile time.
 */
import { createDataFrame, stats as s } from "@tidy-ts/dataframe";

const df = createDataFrame([
  { time: "2014-07-10 11:49:14", amount: 45 },
  { time: "2014-07-10 11:50:14", amount: 45 },
  { time: "2014-07-10 11:51:14", amount: 21 },
]);

// time is string — numeric operations are rejected
// @ts-expect-error — string[] is not assignable to number[]
const wrong = s.mean(df.extract("time"));

// Correct: use time as a string grouping key or parse explicitly
console.log(s.mean(df.extract("amount")));
